pkg/services/folder/folderimpl/unifiedstore.go

```diff
@@ -260,7 +261,10 @@ func (ss *FolderUnifiedStoreImpl) GetFolders(ctx context.Context, q folder.GetFo
 			return nil, fmt.Errorf("unable to convert unstructured item to legacy folder %w", err)
 		}
 		if q.WithFullpath || q.WithFullpathUIDs {
-			parents, err := ss.GetParents(ctx, folder.GetParentsQuery{UID: f.UID, OrgID: q.OrgID})
+			// get parent with service identity
+			// as we the user might not have access to the parent folder
```

Comment:
```suggestion
			// as the user might not have access to the parent folder
```

Reply:
the call to get the folders come from here; https://github.com/grafana/grafana/blob/main/pkg/services/folder/folderimpl/folder_unifiedstorage.go#L66-L67

Using the user.
---

As the question from the legacy code; I am honestly not sure where the original query would be;


Could be here
https://github.com/grafana/grafana/blob/main/pkg/services/folder/folderimpl/dashboard_folder_store.go#L101